prepare.sh

```diff
@@ -4,4 +4,44 @@ set -x
 
 yum -y install git podman
 
-git clone --single-branch --branch=master https://github.com/ceph/ceph-csi /opt/build
+while test $# -gt 0; do
+  case "$1" in
+    -h|--help)
+      echo " "
+      echo "options:"
+      echo "--help|-h                 specify the flags"
+      echo "--ref                     specify the reference of pr"
+      echo "--workdir                 specify the working directory"
+      exit 0
+      ;;
+
+#pr request will be passed through --ref flag
+    --ref*)  
+      export ref=`echo $1 | sed -e 's/^[^=]*=//g'`
+      export ghprbPullId=`echo $ref | cut -d '/' -f2`
+      echo $ref
+      echo ${ghprbPullId}
+      shift
+      ;;
+
+#specifies the directory where repo will be cloned
+      --workdir*)  
+      export workdir=`echo $1 | sed -e 's/^[^=]*=//g'`
+      shift
+      ;;
+    *)
+      break
+      ;;
+  esac
+done
+
+git clone --depth=1 https://github.com/ceph/ceph-csi $workdir
```

Comment:
pass the repository as a parameter to the script too, that makes it possible to run tests from other repositories if that is needed at one point. And it makes it easier to share this script with other CentOS CI projects in the future.